lib/chess/pieces/piece.rb

```diff
@@ -0,0 +1,116 @@
+require_relative '../chess'
+require_relative '../step'
+require 'colorize'
+require 'colorized_string'
+require 'byebug'
+
+module Chess
+  class Piece
+    attr_reader :symbol, :team, :current_step
+
+    Vector = Struct.new(:enabled, :deltas)
+
+    def initialize(position_algebraic, symbol, team, captured = false)
+      @current_step = Step.new(position_algebraic)
```

Comment:
See if we can get rid of the `Step` class, and use `Position` everywhere. Also, every `Position` instance should have a pointer to the previous one